src/ast/mod.rs

```diff
@@ -1363,11 +1363,14 @@ pub enum Statement {
     /// CREATE FUNCTION
     ///
     /// Hive: https://cwiki.apache.org/confluence/display/hive/languagemanual+ddl#LanguageManualDDL-Create/Drop/ReloadFunction
+    /// Postgres: https://www.postgresql.org/docs/15/sql-createfunction.html
     CreateFunction {
+        or_replace: bool,
         temporary: bool,
         name: ObjectName,
-        class_name: String,
-        using: Option<CreateFunctionUsing>,
+        args: Option<Vec<CreateFunctionArg>>,
+        return_type: Option<DataType>,
+        bodies: Vec<CreateFunctionBody>,
```

Comment:
I find the Vec<CreateFunctionBody>` somewhat confusing as it implies that some of them can be repeated. For example, can there be two `CreateFunctionBody::Returning` in this list? I don't think it is possible

Did you consider creating optional fields to encode this in the type system itself?

So instead of 

```rust
    CreateFunction {
        or_replace: bool,
        temporary: bool,
        name: ObjectName,
        args: Option<Vec<CreateFunctionArg>>,
        return_type: Option<DataType>,
        bodies: Vec<CreateFunctionBody>,
```

Something like

```rust

```rust
    CreateFunction {
        or_replace: bool,
        temporary: bool,
        name: ObjectName,
        args: Option<Vec<CreateFunctionArg>>,
        return_type: Option<DataType>,
        as: Option<String>.
        /// LANGUAGE lang_name
        language: Option<Ident>,
...
}
```

?

Reply:
@alamb @wangrunji0408 I have some time, but will not be able to review the whole PR RN, sorry.
- The return type does not match the PostgreSQL syntax. @alamb, do you think it's reasonable to add something like a `SUPPORTED SYNTAX` field to the header of the structure? Here's an [example]( https://github.com/AugustoFKL/sql-helper/blob/dev/src/ansi/ast/common.rs) that I'm implementing for myself.

- @wangrunji0408, for the body structure, it's a problem that we can repeat. You're creating a syntax that can be built wrong for a single dialect. Please let us know if you have a good reason (something like it being repeatable on Snowflake or something like that). However, @alamb, I think having all of them, as you showed, makes building it complex and moving it as well. Sometimes the upstream only wants to handle one or two scenarios, not the whole object. My suggestion is to have a structure that encapsulates them all:
```rust
CREATE FUNCTION {
...,
or_replace: bool, 
temporary: bool, 
name: ObjectName, 
args: Option<Vec<CreateFunctionArg>>, 
return_type: Option<DataType>, 
opt_body: Option<FunctionBody>,
}

pub enum FunctionBody {
Language...
}
```

Reply:
I agree with @AugustoFKL and @alamb regarding repeatable structures. 

From my side, I can still express Hive's `CREATE FUNCTION` syntax so it looks good for me - for me it's okay to change my code when bumping sqlparser-rs dependency.

Reply:
Thanks for this catch. I didn't realize the problem with duplicate items. It has been changed to the way @AugustoFKL mentioned above.